.github/workflows/spellcheck.yml

```diff
@@ -8,8 +8,46 @@ on:
   pull_request:
     types: [opened, synchronize, reopened]
 jobs:
+  list_files:
+    name: List changed files
+    runs-on: ubuntu-22.04
+    outputs:
+      files: ${{ steps.generate-matrix.outputs.files }}
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v2
+        with:
+          # We need all history to list all changed files.
+          fetch-depth: 0
+      - name: Generate matrix
+        id: generate-matrix
+        run: |
+          # List all files that changed between master and head of the
+          # current branch. Encode the output as json.
+          FILES="$(git diff --name-only origin/master..${{ github.event.after }} | jq -cnR '[inputs | select(length>0)]')"
```

Comment:
Don't if this impacts your workflow or not but here, you're storing in an env variable which is limited in terms of capacity. If this is an issue, you should output into a temporary file instead to work around that limitation.